### PR TITLE
Update release checklists post 0.55.2

### PR DESCRIPTION
**Pull request by esc**
as title

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md [hunk 1]
```diff
@@ -17,23 +17,23 @@ labels: task
...
     - [ ] "PR with changelog entries".
 * [ ] Create X.Y release branch.
-* [ ] Pin llvmlite to `>=0.A.0rc1,<0.A+1.0`.
-* [ ] Pin NumPy if needed
-* [ ] Pin tbb if needed
-* [ ] Annotated tag X.Y.Zrc1 on release branch.
+* [ ] Dependency version pinning on release branch
+  * [ ] Pin llvmlite to `>=0.A.0rc1,<0.A+1.0`.
+  * [ ] Pin NumPy if needed
+  * [ ] Pin tbb if needed
```
Comment (stuartarchibald) [inline .github/ISSUE_TEMPLATE/first_rc_checklist.md:23]:
```suggestion
  * [ ] Pin TBB if needed
```

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md [hunk 2]
```diff
@@ -17,23 +17,23 @@ labels: task
 * [ ] Merge change log changes.
     - [ ] "PR with changelog entries".
 * [ ] Create X.Y release branch.
-* [ ] Pin llvmlite to `>=0.A.0rc1,<0.A+1.0`.
-* [ ] Pin NumPy if needed
-* [ ] Pin tbb if needed
-* [ ] Annotated tag X.Y.Zrc1 on release branch.
+* [ ] Dependency version pinning on release branch
```
Comment (stuartarchibald) [inline .github/ISSUE_TEMPLATE/first_rc_checklist.md:20]:
```suggestion
* [ ] Commit build farm configuration and tag with release version.
* [ ] Dependency version pinning on release branch
```
Suggest that the above is added as a reminder.

File: .github/ISSUE_TEMPLATE/sub_rc_checklist.md [hunk 3]
```diff
@@ -18,8 +18,7 @@ labels: task
 * [ ] Build and upload conda packages on buildfarm (check `upload`).
 * [ ] Verify packages uploaded to Anaconda Cloud and move to
   `numba/label/main`.
-* [ ] Build wheels (`$PYTHON_VERSIONS`) on the buildfarm.
-* [ ] Build sdist locally using `python setup.py sdist --owner=ci --group=numba` with umask `0022`.
+* [ ] Build wheels and sdist on the buildfarm (check "upload").
```
Comment (stuartarchibald) [inline .github/ISSUE_TEMPLATE/sub_rc_checklist.md:21]: Suggest adding this:
```
* [ ] Commit build farm configuration and tag with release version.
```
somewhere above before any "building" steps are mentioned.

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch, couple of suggestions else looks good.

File: .github/ISSUE_TEMPLATE/sub_rc_checklist.md (hunk 3 above)
Comment (esc) [reply .github/ISSUE_TEMPLATE/sub_rc_checklist.md:21]: I think it is generally taking a snapshot of the what was used to build the packages is probably a good idea. Though I would propose to add this step to `post-release` steps. In my experience, the farm will often need changes, even after the tag was made. I think that tagging the config version that was used to perform the final buildout is probably closest to what we want.

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md (hunk 1 above)
Comment (esc) [reply .github/ISSUE_TEMPLATE/first_rc_checklist.md]: sure

File: .github/ISSUE_TEMPLATE/sub_rc_checklist.md [hunk 4]
```diff
@@ -29,8 +28,10 @@ labels: task
 
 ### Post release
 
+* [ ] Snapshort Build Farm config
```
Comment (stuartarchibald) [inline .github/ISSUE_TEMPLATE/sub_rc_checklist.md:31]:
```suggestion
* [ ] Snapshot Build Farm config
```

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md [hunk 5]
```diff
@@ -17,23 +17,23 @@ labels: task
...
 * [ ] Upload wheels and sdist to PyPI (upload from `ci_artifacts`).
 * [ ] Verify wheels for all platforms arrived on PyPi.
 * [ ] Initialize and verify ReadTheDocs build.
-* [ ] Clean up `ci_artifacts`.
 * [ ] Send RC announcement email / post announcement to discourse group.
 * [ ] Post link to Twitter.
 
 ### Post Release:
 
+* [ ] Clean up `ci_artifacts` by moving files to subdirectories
```
Comment (stuartarchibald) [inline .github/ISSUE_TEMPLATE/first_rc_checklist.md:36]:
```suggestion
* [ ] Clean up `ci_artifacts` by moving files to sub-directories
```

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md (hunk 2 above)
Comment (esc) [reply .github/ISSUE_TEMPLATE/first_rc_checklist.md:20]: This has been added for the final release checklist. I believe this should suffice. Tagging the buildfarm for a first RC isn't that useful as often changes need to be made for the final!

**Review by sklam — APPROVED** [review]
LGTM

**Pull request merged**